cmd/collectors/restperf/restperf.go

```diff
@@ -0,0 +1,760 @@
+			r.Logger.Error().Stack().Err(err).Msg("set timestamp value: ")
+		}
+		return true
+	})
+
+	if err != nil {
+		r.Logger.Error().Err(err).Msg("Error while processing end points")
+	}
+
+	r.Logger.Info().
+		Uint64("dataPoints", count).
+		Str("apiTime", apiD.String()).
+		Str("parseTime", parseD.String()).
+		Msg("Collected")
+
+	_ = r.Metadata.LazySetValueInt64("api_time", "data", apiD.Microseconds())
+	_ = r.Metadata.LazySetValueInt64("parse_time", "data", parseD.Microseconds())
+	_ = r.Metadata.LazySetValueUint64("count", "data", count)
+	r.AddCollectCount(count)
+
+	// skip calculating from delta if no data from previous poll
+	if r.prop.isCacheEmpty {
+		r.Logger.Debug().Msg("skip postprocessing until next poll (previous cache empty)")
+		r.Matrix = newData
+		r.prop.isCacheEmpty = false
+		return nil, nil
+	}
+
+	calcStart := time.Now()
+
+	r.Logger.Debug().Msg("starting delta calculations from previous cache")
+
+	// cache raw data for next poll
+	cachedData := newData.Clone(true, true, true)
+	// order metrics, such that those requiring base counters are processed last
+	orderedMetrics := make([]matrix.Metric, 0, len(newData.GetMetrics()))
+	orderedKeys := make([]string, 0, len(orderedMetrics))
+
+	for key, metric := range newData.GetMetrics() {
+		if metric.GetName() != "timestamp" {
+			counter := r.prop.counterInfo[metric.GetName()]
+			if counter == nil {
+				// check if it is an array counter
+				counter = r.prop.counterInfo[metric.GetComment()]
+			}
+			if counter != nil {
+				if counter.denominator == "" {
+					// does not require base counter
+					orderedMetrics = append(orderedMetrics, metric)
+					orderedKeys = append(orderedKeys, key)
+				}
+			} else {
+				r.Logger.Warn().Str("counter", metric.GetName()).Msg("Misconfiguration in template counter ")
+			}
+		}
+	}
+
+	for key, metric := range newData.GetMetrics() {
+		if metric.GetName() != "timestamp" {
+			counter := r.prop.counterInfo[metric.GetName()]
+			if counter == nil {
+				// check if it is an array counter
+				counter = r.prop.counterInfo[metric.GetComment()]
+			}
+			if counter != nil {
+				if counter.denominator != "" {
+					// does not require base counter
+					orderedMetrics = append(orderedMetrics, metric)
+					orderedKeys = append(orderedKeys, key)
+				}
+			} else {
+				r.Logger.Warn().Str("counter", metric.GetName()).Msg("Misconfiguration in template counter ")
+			}
+		}
+	}
+
+	// calculate timestamp delta first since many counters require it for postprocessing.
+	// Timestamp has "raw" property, so it isn't post-processed automatically
+	if err = timestamp.Delta(r.Matrix.GetMetric("timestamp")); err != nil {
+		r.Logger.Error().Stack().Err(err).Msg("(timestamp) calculate delta:")
+	}
+
+	var base matrix.Metric
+
+	for i, metric := range orderedMetrics {
+		counter := r.prop.counterInfo[metric.GetName()]
+		if counter == nil {
+			// check if it is an array counter
+			counter = r.prop.counterInfo[metric.GetComment()]
```

Comment:
i used a field comment in metric to figure out if its an array when counter schema is parsed. I will add a new variable in metric struct so that its clear.